includes/Core/Admin/Screens.php

```diff
@@ -408,7 +408,7 @@ private function get_screens() {
 			)
 		);
 
-		$show_splash_in_menu = ! current_user_can( Permissions::VIEW_DASHBOARD ) && ! current_user_can( Permissions::VIEW_MODULE_DETAILS ) && ! current_user_can( Permissions::MANAGE_OPTIONS );
+		$show_splash_in_menu = current_user_can( Permissions::VIEW_SPLASH ) && ! current_user_can( Permissions::VIEW_DASHBOARD );
```

Comment:
This reads quite a bit more intuitively now and makes the condition quite clear. Without this change, the splash page will not appear in the admin menu when a non-admin has the ability to `VIEW_SPLASH` when `dashboardSharing` is enabled.

This is a change outside the feature flag though so it should be thoroughly checked. In my testing, it seems to cover all the cases both the feature on and off.